cloudevents-sdk-rdkafka/src/kafka_consumer_record.rs

```diff
@@ -0,0 +1,264 @@
+
+        resp_des
+    }
+
+    pub fn borrowed_new(message: &BorrowedMessage) -> ConsumerRecordDeserializer {
+        let mut resp_des = ConsumerRecordDeserializer {
+            headers: HashMap::new(),
+            payload: Bytes::new(),
+        };
+        let headers = message.headers().unwrap();
+        for i in 0..headers.count() {
+            let header = headers.get(i).unwrap();
+            resp_des
+                .headers
+                .insert(header.0.to_string(), Bytes::copy_from_slice(header.1));
+        }
+
+        match message.payload() {
+            Some(s) => resp_des.payload = Bytes::copy_from_slice(s),
+            None => resp_des.payload = resp_des.payload,
+        }
+
+        resp_des
+    }
+}
+
+impl BinaryDeserializer for ConsumerRecordDeserializer {
+    fn deserialize_binary<R: Sized, V: BinarySerializer<R>>(self, mut visitor: V) -> Result<R> {
+        if self.encoding() != Encoding::BINARY {
+            return Err(message::Error::WrongEncoding {});
+        }
+
+        let spec_version = SpecVersion::try_from(header_value_to_str!(self
+            .headers
+            .get("ce_specversion")
+            .unwrap())?)?;
+
+        visitor = visitor.set_spec_version(spec_version.clone())?;
+
+        let attributes = spec_version.attribute_names();
+
+        for (hn, hv) in self
+            .headers
+            .iter()
+            .filter(|(hn, _)| "ce_specversion" != **hn && hn.starts_with("ce_"))
+        {
+            let name = &hn["ce_".len()..];
+
+            if attributes.contains(&name) {
+                visitor = visitor.set_attribute(
+                    name,
+                    MessageAttributeValue::String(String::from(header_value_to_str!(hv)?)),
+                )?
+            } else {
+                visitor = visitor.set_extension(
+                    name,
+                    MessageAttributeValue::String(String::from(header_value_to_str!(hv)?)),
+                )?
+            }
+        }
+
+        if let Some(hv) = self.headers.get("content-type") {
+            visitor = visitor.set_attribute(
+                "datacontenttype",
+                MessageAttributeValue::String(String::from(header_value_to_str!(hv)?)),
+            )?
+        }
+
+        if self.payload.len() != 0 {
+            visitor.end_with_data(self.payload.to_vec())
+        } else {
+            visitor.end()
+        }
+    }
+}
+
+impl StructuredDeserializer for ConsumerRecordDeserializer {
+    fn deserialize_structured<R: Sized, V: StructuredSerializer<R>>(self, visitor: V) -> Result<R> {
+        if self.encoding() != Encoding::STRUCTURED {
+            return Err(message::Error::WrongEncoding {});
+        }
+        visitor.set_structured_event(self.payload.to_vec())
+    }
+}
+
+impl MessageDeserializer for ConsumerRecordDeserializer {
+    fn encoding(&self) -> Encoding {
+        match (
+            str::from_utf8(self.headers.get("content-type").unwrap_or(&Bytes::new()))
```

Comment:
This condition can be simplified with a `map` step. Given the change about `Vec<u8>`, you can do something like `self.headers.get("content-type").map(|s| String::from_utf8(s).ok()).flatten().map(|s| s.starts_with(...))`